src/NumericalAlgorithms/LinearOperators/PowerMonitors.cpp

```diff
@@ -48,9 +49,69 @@ void power_monitors(const gsl::not_null<std::array<DataVector, Dim>*> result,
 
 template <size_t Dim>
 std::array<DataVector, Dim> power_monitors(
-    const DataVector& input_data_vector, const Mesh<Dim>& mesh) {
+    const DataVector& data_vector, const Mesh<Dim>& mesh) {
   std::array<DataVector, Dim> result{};
-  power_monitors(make_not_null(&result), input_data_vector, mesh);
+  power_monitors(make_not_null(&result), data_vector, mesh);
+  return result;
+}
+
+// The power_monitor argument should be made of type ModalVector
+// when pybindings for ModalVector are enabled
+double relative_truncation_error(const DataVector& power_monitor,
+                                 const size_t num_modes_to_use) {
+  ASSERT(
+      num_modes_to_use <= power_monitor.size(),
+      "Number of modes needs less or equal than the number of power monitors");
+  ASSERT(2_st <= num_modes_to_use,
+         "Number of modes needs to be larger or equal than 2.");
+  // Compute weighted average and total sum in the current dimension
+  double weighted_average = 0.0;
+  double weight_sum = 0.0;
+  double weight_value = 0.0;
+  const size_t last_index = num_modes_to_use - 1;
+  for (size_t index = 0; index <= last_index; ++index) {
+    // Compute current weight
+    weight_value =
+        exp(-square(index - static_cast<double>(last_index) + 0.5));
+    // Add weighted power monitor
+    const double mode = power_monitor[index];
+    ASSERT(not equal_within_roundoff(mode, 0.0),
+           "A power monitor is zero within roundoff error. If this is one of "
+           "the highest modes, reduce the number of power monitors used to "
+           "estimate the relative truncation error.");
+    weighted_average += weight_value * log10(mode);
+    // Add term to weighted sum
+    weight_sum += weight_value;
+  }
+  weighted_average /= weight_sum;
+
+  // Maximum between the first two power monitors
+  double leading_term = std::max(power_monitor[0], power_monitor[1]);
+  ASSERT(not equal_within_roundoff(leading_term, 0.0),
+         "The leading power monitor term is zero within roundoff error.");
+  leading_term = log10(leading_term);
+
+  // Compute relative truncation error
+  return leading_term - weighted_average;
+}
+
+template <size_t Dim>
+std::array<double, Dim> truncation_error(
+    const std::array<DataVector, Dim>& power_monitors,
+    const DataVector& data_vector) {
```

Comment:
While I agree that this is technically a `DataVector`, can you give this a more descriptive name like `solution` or `variable` or something? Even `u` is better since `u` is often shorthand for the evolved vars. You can then name the first argument something like `power_monitors_of_u`